main.cpp

```diff
@@ -227,6 +454,32 @@ struct CarWash //                                   1) define an empty struct fo
     3) define waveform with adsr
  */
 
+struct Lfo       
+{
+    //has rate
+    int rate = 127; //3) member variables with relevant data types.
+    //has amount
+    int amount = 127; //3) member variables with relevant data types.
+    //has waveform
+    int waveform = 127; //3) member variables with relevant data types.
+    //has adsr
+    int adsr = 127; //3) member variables with relevant data types.
+    //has retrigger
+    int retrigger = 127; //3) member variables with relevant data types.
+
+    /* 
+    member functions with a user-defined type as the parameter.
+    The user-defined type parameter happens to be the nested class.
+    */
+
+    //2) create waveform
+    void createWaveform();
+    //2) controll other parameters with waveform
+    void sendWaveform();
+    //2) define waveform with adsr
+    void defineWaveformAdsr();
```

Comment:
Definitely need parameters here.   what waveform is being created?   Where is the waveform being sent?   
What are the ADSR parameters that are being used to create the waveform ADSR? 
- [ ] revise